### PR TITLE
WIP:  Incorporate ITK adaptive denoising module.

### PR DESCRIPTION
**Pull request by ntustison**

### CODE REVIEW

File: SuperBuild/External_ITKv5.cmake [hunk 1]
```diff
@@ -151,8 +152,8 @@ if(NOT DEFINED ${extProjName}_DIR AND NOT ${USE_SYSTEM_${extProjName}})
   ### --- End Project specific additions
   set(${proj}_REPOSITORY ${git_protocol}://github.com/InsightSoftwareConsortium/ITK.git)
   # set(${proj}_REPOSITORY ${git_protocol}://github.com/stnava/ITK.git)
-  set(${proj}_GIT_TAG  1e708db2f586997e408cfdc2cea5114ae5575892)  # Type -> Enum  Update to ITKv5rc02 2020-03-18
-  set(ITK_VERSION_ID ITK-5.1) ### NOTE: When updating GIT_TAG, also update ITK_VERSION_ID
+  set(${proj}_GIT_TAG e21a56d1227c5433066237060368cb4532b8a9d2)  # Adaptive denoising module 10/2/2020
+  set(ITK_VERSION_ID ITK-5.1.1) ### NOTE: When updating GIT_TAG, also update ITK_VERSION_ID
```
Comment (cookpa) [inline SuperBuild/External_ITKv5.cmake:156]: I had the same build error when I tried to do this exact thing. For some reason CMake fails to understand 5.1.1 but it likes 5.1

**Comment by cookpa** [comment]
Hi @ntustison I had the same error here when building another branch. CMake doesn't seem to understand ITK version "5.1.1". I changed it back to 5.1 and it compiled as normal.

Will these changes alter the performance of DenoiseImage?

One other note: updating ITK will switch the NIFTI I/O with Hans' recent changes to prefer the sform. I think this is a good change and should reduce the inputs not in the same physical space errors. But it will require some documentation updates. I've been meaning to test it and will make it a priority to re-run my DTI tests once this is merged. 

Might be time for a new release once this is merged?

**Comment by ntustison** [comment]
Thanks @cookpa --- that's good to know.  I'll change that.

A new release sounds like a great idea.  We need to start getting on some sort of schedule for the grant anyway.

File: SuperBuild/External_ITKv5.cmake [hunk 2]
```diff
@@ -153,7 +153,7 @@ if(NOT DEFINED ${extProjName}_DIR AND NOT ${USE_SYSTEM_${extProjName}})
   set(${proj}_REPOSITORY ${git_protocol}://github.com/InsightSoftwareConsortium/ITK.git)
   # set(${proj}_REPOSITORY ${git_protocol}://github.com/stnava/ITK.git)
   set(${proj}_GIT_TAG e21a56d1227c5433066237060368cb4532b8a9d2)  # Adaptive denoising module 10/2/2020
-  set(ITK_VERSION_ID ITK-5.1.1) ### NOTE: When updating GIT_TAG, also update ITK_VERSION_ID
+  set(ITK_VERSION_ID ITK-5.1) ### NOTE: When updating GIT_TAG, also update ITK_VERSION_ID
```
Comment (cookpa) [inline SuperBuild/External_ITKv5.cmake:156]: Circleci is still failing with the same error. I compiled locally to debug and CMake appears to want ITK_VERSION_ID to be ITK-5.2.

ITK-5.1 worked for me but I was building the actual 5.1.1 release tag. Your commit is ahead of that. It's like it's rounding up somehow.

Comment (ntustison) [reply SuperBuild/External_ITKv5.cmake:156]: Thanks @cookpa .  So what would you recommend?  Should we simply label it as 5.2?

Comment (cookpa) [reply SuperBuild/External_ITKv5.cmake:156]: That is the immediate fix. I can try to understand it further.

What happens is that CMake writes

`staging/lib/cmake/ITK-5.2/ITKConfig.cmake`

even when the file contains `set(ITK_VERSION_ID ITK-5.1)`.

**Comment by ntustison** [comment]
Thanks Phil.

**Comment by cookpa** [comment]
Looks like a typo found its way in

```
-- SuperBuild - ANTS => Requires ITKv5, 
CMake Error at SuperBuild/External_ITKv5.cmake:156:
  Parse error.  Function missing ending ")".  End of file reached.
Call Stack (most recent call first):
  CMake/SlicerMacroCheckExternalProjectDependency.cmake:69 (include)
  SuperBuild.cmake:224 (SlicerMacroCheckExternalProjectDependency)
  CMakeLists.txt:67 (include)
```

**Comment by cookpa** [comment]
Wait: I just noticed this is tagged WIP. Is it ready to merge @ntustison?

**Comment by ntustison** [comment]
Yeah, it should be good to go.  Thanks.

**Pull request merged**